### DISCUSSION TITLE
Disable "Report a security vulnerability" option/button?

### DISCUSSION THREAD

**Discussion by timwoj** [community/community#45567] [Projects and Issues] [answered]
### Select Topic Area

Question

### Body

I'm finally getting around to adding some issue templates to one of our repositories. Is it possible to disable the "Report a security vulnerability" option that gets automatically added to the list of templates when a user tries to create a new issue? My team already has external ways to report those sorts of issues, and so we don't use Github for them. Is there an option in the `config.yml` schema that disables it? If not, can one be added?

**Comment by timwoj** [comment]
Thinking about this a bit more, the repository I'm tinkering with this on doesn't have security reporting enabled. Could it possibly tie into that setting? If you want the button, turn that on, otherwise don't show it?

**Reply by JasonLandbridge** [reply]
I posted the answer/solution [here](https://github.com/orgs/community/discussions/45567#discussioncomment-5313960)!

**Comment by jhaaaa** [comment]
I was actually surprised to see the Report a vulnerability button present in my issues panel:

![image](https://user-images.githubusercontent.com/5481259/217985844-a77cff5f-4e6a-4019-a4e7-a2ed8b9f6460.png)

Because the [docs make it sound like it wouldn't display](https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/configuring-private-vulnerability-reporting-for-a-repository) unless I enabled "Private vulnerability reporting", which I haven't:
![image](https://user-images.githubusercontent.com/5481259/217985629-d3c53b3a-8680-476e-a8a5-a4d5e2f68fa5.png)

Or is the Report a vulnerability button I'm seeing different from what would be enabled by "private vulnerability reporting"?

I've enabled and disabled the feature and the button remains displayed.

Any thoughts on this?

Thank you!

p.s. I just found this other location to Enable vulnerability reporting:

![image](https://user-images.githubusercontent.com/5481259/218142079-8b99e4da-f261-4ab6-a85d-0ac3b0712677.png)

**Reply by JasonLandbridge** [reply]
I posted the answer/solution [here](https://github.com/orgs/community/discussions/45567#discussioncomment-5313960)!

**Comment by zhou-tao** [comment]
I also encountered the same problem, is there a solution?

**Reply by JasonLandbridge** [reply]
I posted the answer/solution [here](https://github.com/orgs/community/discussions/45567#discussioncomment-5313960)!

**Comment by raiguard** [comment]
This has appeared in all of my repos as I have been migrating back to GitHub. I mostly make [Factorio mods](https://github.com/raiguard/BottleneckLite/issues/new/choose), and reporting security concerns on those is entirely pointless, as mods are completely sandboxed. How can I remove this option?

**Reply by JasonLandbridge** [reply]
I posted the answer/solution [here](https://github.com/orgs/community/discussions/45567#discussioncomment-5313960)!

**Comment by Elliria** [comment]
This is something I'll be following with great interest, because I've seen repositories that have managed to remove it and would like to do so, too.

The official GitHub instructions for [Enabling or disabling private vulnerability reporting for a repository](https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/configuring-private-vulnerability-reporting-for-a-repository#enabling-or-disabling-private-vulnerability-reporting-for-a-repository) do not work for me. That button is still there when creating a new issue.

These steps should work for disabling private vulnerability reporting in all of your repositories, but they don't work for me, either:
1. Click on your GitHub profile picture.
2. Click on **Settings** in the drop-down menu.
3. Scroll down to the **Security** section in the left panel.
4. Click on **Code security and analysis** in that section in the left panel.
5. Click the "Disable all" button in the **Private vulnerability reporting** section in the right panel.
6. Read the warning that pops up.
7. Click the **Disable private vulnerability reporting** button.
8. Clear your browser's cache and cookies.

I hope someone will come up with steps that work.

**Reply by JasonLandbridge** [reply]
I posted the answer/solution [here](https://github.com/orgs/community/discussions/45567#discussioncomment-5313960)!

**Comment by JasonLandbridge** [comment]
I found the answer, it's only administrators who see it!

Source: https://twitter.com/adamwathan/status/1617171385800957954

If you go to my [PlexRipper issue chooser](https://github.com/PlexRipper/PlexRipper/issues/new/choose) you will not see it because you're not admin. If you do see it, lol how?

Admin view:
![image](https://user-images.githubusercontent.com/15127381/225117780-26e511a3-4dab-402b-92dd-dfa6d017201a.png)

Normal user view:
![image](https://user-images.githubusercontent.com/15127381/225118085-4c55db61-6c6b-4bf2-a187-d293df8565dc.png)

**Reply by JasonLandbridge** [reply]
@timwoj, if this answer is correct could you mark this as the accepted answer?

**Reply by Elliria** [reply]
Thank you so much! That solves it for me.

**Reply by guylepage3** [reply]
This additional button should be "Disabled" by default with the ability to "Enable" it in the Security settings. This is OVERKILL for users.

A Product Design fundamental...

```
Give the user the option but don't force things upon them.
```

**Reply by guylepage3** [reply]
Furthermore.. Users will most likely disable security features as a work around to remove the "non-optional" button. I know myself and my teammates will be doing so.

**Reply by JasonLandbridge** [reply]
So then complain at GitHub, why down vote all my answers here?

**Reply by Elliria** [reply]
It's not for users. If you look at the two screenshots just above this, you'll see that the users don't see it. Only administrators see it.

**Reply by guylepage3** [reply]
>So then complain at GitHub, why down vote all my answers here?
@JasonLandbridge Yes sir. I created a support ticket.

>It's not for users. If you look at the two screenshots just above this, you'll see that the users don't see it. Only administrators see it.
@Elliria. Admins are users of GitHub the last I checked.

**Reply by Elliria** [reply]
That's true. It would definitely be nice of them to make it optional entirely, but I'm just glad that it at least doesn't show up for visitors who aren't part of the project.

**Reply by Elliria** [reply]
Also, it's only for administrators of the project in question. When they're out and about on GitHub, they won't see it on other projects.

**Comment by gingerbeardman** [comment]
I also have this problem. It is visible to people other than myself. "Solution" posted is not a solution as far as I can see.

**Reply by Elliria** [reply]
It should only be visible to people on your team. If you log out and start to create an issue report in your repository, do you still see it or is it gone?

**Reply by gingerbeardman** [reply]
I don't have a team, but I asked a friend to check my repo and they see it too.

https://github.com/gingerbeardman/greatwavetoday

I can't log out and try it as it says I need to be logged into GitHub to file an issue.

![image](https://user-images.githubusercontent.com/49612/230898518-dfcbbfa4-8d12-441c-8248-535ff0cec6e0.png)

**Reply by JasonLandbridge** [reply]
This is what I see when I create an issue, so the solution applies: 
![image](https://user-images.githubusercontent.com/15127381/230898890-124333c3-68ae-4959-9321-2ea21a2b75a0.png)

**Reply by gingerbeardman** [reply]
Good to know thanks, so why did my friend see it?

**Reply by JasonLandbridge** [reply]
Haha you're asking me what you friend saw? How would I know :P Ask him for a screenshot or maybe he has write/admin access to the repo

**Reply by gingerbeardman** [reply]
He sent me a screenshot, he could see it, he does not have access to this repo but is Collaborator on another of my repos (not an admin).

**Reply by Elliria** [reply]
This is what I see:

![WhatISee](https://user-images.githubusercontent.com/25792925/230935810-4ca15a1f-0bd3-4482-afbc-5baae773a997.png)

**Comment by guylepage3** [comment]
If it's viewable to your team, it's an issue as they are the ones that see it most.

**Comment by Elliria** [comment]
True. I don't like it, either, and would like to see it made optional.

**Comment by KateCatlin** [comment, marked as answer]
Hey all! 

First of all, I'm the Product Manager for advisories at GitHub and I apologize for this issue. 

**This was a bug and it is now fixed!** 

Before this change, if you had a `security policy` written, the "see security policy" option on the issue choose was always shown. Now, if you have Private Vulnerability Reporting enabled, we show that instead. But we mistakenly were forcing the private vulnerability reporting line on all admins, whether or not you had PVR enabled. 

This should now be resolved! Please chime in if it's still not working for you.

**Reply by Elliria** [reply]
Thank you so much. Problem solved over here.

**Reply by gingerbeardman** [reply]
Solved here also! Thanks

**Reply by timwoj** [reply]
Great, thanks!

**Comment by jackiekazil** [comment]
Unless I am missing this - this doesn't seem to fixed... Reviewing this doc -  https://docs.github.com/en/code-security/security-advisories/working-with-repository-security-advisories/configuring-private-vulnerability-reporting-for-a-repository 

In my admin panel - there is no "Disable" button. 
Please advise.
